File: common/src/main/java/org/sandboxpowered/silica/block/SilicaBlockState.java
package org.sandboxpowered.silica.block;

import com.google.common.collect.ImmutableMap;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.property.Property;
import org.sandboxpowered.api.util.math.Position;
import org.sandboxpowered.api.world.World;
import org.sandboxpowered.silica.state.BaseState;

public class SilicaBlockState extends BaseState<Block, BlockState> implements BlockState {
    public SilicaBlockState(Block base, ImmutableMap<Property<?>, Comparable<?>> properties) {
        super(base, properties);
    }

    @Override
    public Block getBlock() {
        return base;
    }

    @Override
    public float getDestroySpeed(World world, Position pos) {
        return 1;
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/state/block/SilicaBlockState.kt
package org.sandboxpowered.silica.state.block

import com.google.common.collect.ImmutableMap
import org.sandboxpowered.api.block.Block
import org.sandboxpowered.api.world.state.BlockState
import org.sandboxpowered.api.world.state.Property
import org.sandboxpowered.silica.state.BaseState

class SilicaBlockState(base: Block, properties: ImmutableMap<Property<*>, Comparable<*>>) :
    BaseState<Block, BlockState>(base, properties), BlockState {
    override fun getBlock(): Block {
        return base
    }
}